### PR TITLE
Align configurations defaults between default file and Java object (broker.conf, proxy.conf, websocket.conf)

### PR DESCRIPTION
**Pull request by nicoloboschi**
Based on #13268 
### Motivation

In the default configuration files `conf/broker.conf`, `conf/proxy.conf` and `conf/websocket.conf` some entries are set with a value which differs from the Java one.

So if you comment out or you remove that line in the configuration file, the behavior changes.

### Modifications

* Aligned some Java properties to the file's value. Likely the user will not comment out the config lines so the expected behavior is that the config file contains the default values.
* Added tests to avoid future regressions 

#### Broker

| Property | broker.conf | Java | New value |
| --------- | ----------- | ----- | ---------- |
|  brokerDeduplicationSnapshotFrequencyInSeconds | 10s | 120s | 120s |  
|  numWorkerThreadsForNonPersistentTopic | 8 | availableProcessors | availableProcessors |  
|  managedLedgerNumWorkerThreads | 8 | availableProcessors | availableProcessors |  
|  webSocketNumIoThreads | 8 | availableProcessors | availableProcessors |  
|  webSocketConnectionsPerBroker | 8 | availableProcessors | availableProcessors |  
|  topicPublisherThrottlingTickTimeMillis | 10 | 5 | 10 |
|  dispatchThrottlingOnNonBacklogConsumerEnabled | true | false | true (see #1654) |
|  replicationMetricsEnabled | true | false | true |  

#### WebSocket
| Property | websocket.conf | Java | New value |
| --------- | ---------------- | ----- | ---------- |  
|  bindAddress | 0.0.0.0 | null | 0.0.0.0 |  
|  webSocketNumIoThreads | 8 | availableProcessors | availableProcessors |  
|  webSocketConnectionsPerBroker | 8 | availableProcessors | availableProcessors |  

### Does this pull request potentially affect one of the following parts:

 - The default values of configurations: yes
 
### Documentation

- [x] `doc`

### CODE REVIEW

**Comment by nicoloboschi** [comment]
@eolivelli @codelipenghui PTAL

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java [hunk 1]
```diff
@@ -933,7 +933,7 @@
     @FieldContext(
         category = CATEGORY_SERVER,
         doc = "Number of worker threads to serve non-persistent topic")
-    private int numWorkerThreadsForNonPersistentTopic = Runtime.getRuntime().availableProcessors();
+    private int numWorkerThreadsForNonPersistentTopic = 8;
```
Comment (merlimat) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java:936]: This is done to use the number of available core. We must not change it.

**Review by merlimat — CHANGES_REQUESTED** [review]
This is changing a lot of defaults which must not be changed.

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java [hunk 2]
```diff
@@ -576,7 +576,7 @@
         category = CATEGORY_POLICIES,
         doc = "How often is the thread pool scheduled to check whether a snapshot needs to be taken.(disable with value 0)"
     )
-    private int brokerDeduplicationSnapshotFrequencyInSeconds = 120;
+    private int brokerDeduplicationSnapshotFrequencyInSeconds = 10;
```
[the remark this answers is not in the document]
Comment (nicoloboschi) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java:579]: This is value set in the broker.conf so it is the value used by default. The goal of this patch is not providing better defaults but to keep consistency if a user remove a configuration line. It is expected the values in the broker.conf are the default ones and not an "improved configuration". Do you agree?

[the remark this answers is not in the document]
Comment (merlimat) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java:579]: We need to fix these, but we need to fix them with default that make sense.

**Comment by nicoloboschi** [comment]
@merlimat I addressed your comments. I also added a table in the PR description to better understand the configuration changes

**Review by merlimat — APPROVED** [review]
👍

**Pull request merged**